subxt/src/backend/legacy/mod.rs

```diff
@@ -181,20 +181,16 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for LegacyBackend<T> {
 
     async fn current_runtime_version(&self) -> Result<RuntimeVersion, Error> {
         let details = self.methods.state_get_runtime_version(None).await?;
-        Ok(RuntimeVersion {
-            spec_version: details.spec_version,
-            transaction_version: details.transaction_version,
-        })
+        Ok(RuntimeVersion::new(
+            details.spec_version,
```

Comment:
The only thing I don't reallt like about `RuntimeVersion` becoming a private struct with accessors etc is that because `spec_version` and `transaction_version` are actually the same type, it's now a little easier to accidentally mix them up here.

Reply:
yeah agree, maybe we could make them pub and #[doc(hidden)] instead.

Because it's in subxt-core it won't work unless these are fields public